Utils/HandlerUtil.py

```diff
@@ -61,26 +61,62 @@
 import base64
 import json
 import time
+
+from xml.etree import ElementTree
 from os.path import join
 from Utils.WAAgentUtil import waagent
 from waagent import LoggerInit
 
 DateTimeFormat = "%Y-%m-%dT%H:%M:%SZ"
 
+MANIFEST_XML = "manifest.xml"
+
 class HandlerContext:
     def __init__(self,name):
         self._name = name
         self._version = '0.0'
         return
 
 class HandlerUtility:
-    def __init__(self, log, error, short_name):
+    def __init__(self, log, error, s_name=None, l_name=None, version=None):
         self._log = log
         self._error = error
-        self._short_name = short_name
 
+        if s_name is None or l_name is None or version is None:
+            (long_name, short_name, ext_version) = self._get_extension_info()
+            self._short_name = short_name
+            self._extension_version = ext_version
+            self._log_prefix = '[%s-%s] ' % (long_name, ext_version)
```

Comment:
I've had no other motivation than fixing the wrong extension version in logs (1.0) to the correct one (e.g., 2.3.9001), and retaining all other existing behaviors. The long name has been used for LAD for so long, and current LAD E2E test suite depends on the long name, which can be changed, of course, so I'm OK with changing it to short name.